webapp/src/components/rhs/badge_details.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+
+import Scrollbars from 'react-custom-scrollbars';
+
+import {BadgeDetails, BadgeID} from '../../types/badges';
+import Client from '../../client/api';
+
+import {RHSState} from '../../types/general';
+import {RHS_STATE_MY, RHS_STATE_OTHER} from '../../constants';
+import BadgeImage from '../utils/badge_image';
+
+import {markdown} from 'utils/markdown';
+
+import UserRow from './user_row';
+
+import './badge_details.scss';
+
+type Props = {
+    badgeID: BadgeID | null;
+    currentUserID: string;
+    actions: {
+        setRHSView: (view: RHSState) => void;
+        setRHSUser: (user: string | null) => void;
+    };
+}
+
+type State = {
+    loading: boolean;
+    badge?: BadgeDetails | null;
+}
+
+function renderView(props: any) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--view'
+        />);
+}
+
+function renderThumbHorizontal(props: any) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--horizontal'
+        />);
+}
+
+function renderThumbVertical(props: any) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--vertical'
+        />);
+}
+
+class BadgeDetailsComponent extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+        };
+    }
+
+    componentDidMount() {
+        if (this.props.badgeID === null) {
+            return;
+        }
+
+        const c = new Client();
+        c.getBadgeDetails(this.props.badgeID).then((badge) => {
+            this.setState({badge, loading: false});
+        });
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        // Typical usage (don't forget to compare props):
+        if (this.props.badgeID === prevProps.badgeID) {
+            return;
+        }
+
+        if (this.props.badgeID === null) {
+            return;
```

Comment:
should it be `this.setState({badge: null, loading: false})`

Reply:
Nevermind, I saw later that if there is null badge and loading is not relevant at all.